tests/components/pi_hole/test_sensor.py

```diff
@@ -0,0 +1,14 @@
+"""Test pi_hole sensor."""
+
+from homeassistant.components.pi_hole import PiHoleData
+from homeassistant.components.pi_hole.sensor import PiHoleSensor, SENSOR_DICT
+from unittest.mock import patch
+
+
+def test_sensor_class_init(hass):
+    """Test that a sensot is constructed correctly."""
+    data = PiHoleData(None, "Test Pi-Hole")
+    with patch.dict(SENSOR_DICT, {"test": ["Test", "widgets", "mdi:test"]}):
+        sensor = PiHoleSensor(data, "test")
```

Comment:
Please don't interact with the entity directly. Check the state and state attributes of the entity. Set up the component and platform and assert the state after the entity has been added.

We test this in the test above, so maybe just extend that test?

Reply:
My intent is to test that the customization options for a sensor are honored. I'm indirectly testing this in the integration init test, so i'll just add the relevant assertions there